### PR TITLE
cclient: added extended API header file.

### PR DESCRIPTION
**Pull request by oopsmonk**
This PR reviews extended API design in cclient, those extended API functions refer to the [iota/core](https://github.com/iotaledger/iota.js/tree/next/packages/core) in `iota.js`.
`TODO` objects and object operations should be designed and implemented during the API implementation.

# Test Plan:
No

### CODE REVIEW

File: cclient/iota_extended_api.h [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+typedef UT_array input_list_t;
+
+typedef struct {
+  iota_http_service_t http;
+  serializer_t serializer;
+} iota_service_t;
+
+typedef struct {
+  int64_t balance;
+  int32_t keyIndex;
```
Comment (tsvisabo) [inline cclient/iota_extended_api.h]: snake_case

**Review by tsvisabo — CHANGES_REQUESTED** [review]

File: cclient/iota_extended_api.h [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+extern "C" {
+#endif
+
+//----TODO objects----------
+typedef char_buffer_t hash_t;
+typedef UT_array hashes_t;
+typedef UT_array address_list_t;
+typedef UT_array inputs_list_t;
+typedef UT_array transfer_list_t;
+typedef UT_array transaction_list_t;
```
Comment (tsvisabo) [inline cclient/iota_extended_api.h]: that definitions are useful for other cases, can you have them in `common`?

**Review by tsvisabo — CHANGES_REQUESTED** [review]

File: cclient/iota_extended_api.h (hunk 2 above)
Comment (oopsmonk) [reply cclient/iota_extended_api.h]: it's in the plan, most of those objects should be in the `common`.  :wink:

**Comment by thibault-martinez** [comment]
A general comment about the naming of these functions.
Maybe we want to reserve the `iota_api_*` scope for the actual cIRI API ?
What about `iota_client_*` ?

**Comment by oopsmonk** [comment]
> A general comment about the naming of these functions.
> Maybe we want to reserve the `iota_api_*` scope for the actual cIRI API ?
> What about `iota_client_*` ?

`iota_client_*` sounds good for cclient, but how about  `iota_iri_*` for cIRI since `com.iota.iri.*` packages in IRI?

File: cclient/iota_extended_api.h [hunk 3]
```diff
@@ -0,0 +1,429 @@
...
+  transfer_list_t transfers;
+  transaction_list_t transactions;
+} account_data_t;
+
+typedef struct {
+  int security;
+  int32_t start;
+  int32_t end;
+} address_opt_t;
+//---------END of TODO objects-------------
```
Comment (th0br0) [inline cclient/iota_extended_api.h]: Can we get rid of the TODO here?

**Review by th0br0 — APPROVED** [review]

**Comment by th0br0** [comment]
We could also just just not have a `iota_` prefix? :thinking:

**Comment by thibault-martinez** [comment]
We definitely need to have a naming convention discussion :)

**Comment by oopsmonk** [comment]
> We could also just just not have a `iota_` prefix? thinking

do you mean both of file name and API functions?

**Review by tsvisabo — APPROVED** [review]

**Pull request merged**